app/validation/validator.py

```diff
@@ -149,8 +155,12 @@ def validate_routing_on_answer_options(self, block, answer):
             for rule in block['routing_rules']:
                 if 'goto' in rule and 'when' in rule['goto'].keys():
                     when = rule['goto']['when']
-                    if 'id' in when and when['id'] == answer['id'] and when['value'] in options:
-                        options.remove(when['value'])
+                    if 'id' in when[0]:
```

Comment:
Looks like this was already broken. But this is explicitly only check the first `when` rule. Could you iterate through the `when` list and check each rule in it instead of just grabbing the first one?